From 3e8f83d40c6fdd89653ef547aa16cffe1f3c1793 Mon Sep 17 00:00:00 2001
From: aleks <aleks.stier@icloud.com>
Date: Sun, 4 Aug 2019 13:23:10 +0200
Subject: [PATCH] Apply nord-theme

*"Inspired by the beauty of the arctic, the colors reflect the cold, yet
harmonious world of ice and the colorfulness of the Aurora Borealis."* - [Nord
Theme](https://www.nordtheme.com/)

There are also many [ports](https://www.nordtheme.com/ports) for other programs
like [vim](https://www.nordtheme.com/ports/vim) and
[tmux](https://www.nordtheme.com/ports/tmux) to make the overall appearance
coherent. I would recommend to use it in combination with the arc-theme for gtk
(fits perfectly).

The default behaviour of st is to reverse the fore- and background colors of
each selected cell. If you want that the selection-colors are not reveresed but
instead have fixed fore- and background colors apply on top of this patch the
[selectioncolors](../selectioncolors/)-patch. Then set the following settings
in your config.h:

static unsigned int defaultcs = 257;
static unsigned int defaultrcs = 257;
unsigned int selectionfg = 257;
unsigned int selectionbg = 0;
---
 config.def.h | 44 ++++++++++++++++++++++----------------------
 1 file changed, 22 insertions(+), 22 deletions(-)

diff --git a/config.def.h b/config.def.h
index 0e01717..fb504d4 100644
--- a/config.def.h
+++ b/config.def.h
@@ -85,30 +85,30 @@ unsigned int tabspaces = 8;
 /* Terminal colors (16 first used in escape sequence) */
 static const char *colorname[] = {
 	/* 8 normal colors */
-	"black",
-	"red3",
-	"green3",
-	"yellow3",
-	"blue2",
-	"magenta3",
-	"cyan3",
-	"gray90",
+	"#3b4252", /* black   */
+	"#bf616a", /* red     */
+	"#a3be8c", /* green   */
+	"#ebcb8b", /* yellow  */
+	"#81a1c1", /* blue    */
+	"#b48ead", /* magenta */
+	"#88c0d0", /* cyan    */
+	"#e5e9f0", /* white   */
 
 	/* 8 bright colors */
-	"gray50",
-	"red",
-	"green",
-	"yellow",
-	"#5c5cff",
-	"magenta",
-	"cyan",
-	"white",
+	"#4c566a", /* black   */
+	"#bf616a", /* red     */
+	"#a3be8c", /* green   */
+	"#ebcb8b", /* yellow  */
+	"#81a1c1", /* blue    */
+	"#b48ead", /* magenta */
+	"#8fbcbb", /* cyan    */
+	"#eceff4", /* white   */
 
 	[255] = 0,
 
 	/* more colors can be added after 255 to use with DefaultXX */
-	"#cccccc",
-	"#555555",
+	"#2e3440", /* background */
+	"#d8dee9", /* foreground */
 };
 
 
@@ -116,10 +116,10 @@ static const char *colorname[] = {
  * Default colors (colorname index)
  * foreground, background, cursor, reverse cursor
  */
-unsigned int defaultfg = 7;
-unsigned int defaultbg = 0;
-static unsigned int defaultcs = 256;
-static unsigned int defaultrcs = 257;
+unsigned int defaultfg = 257;
+unsigned int defaultbg = 256;
+static unsigned int defaultcs = 257;
+static unsigned int defaultrcs = 256;
 
 /*
  * Default shape of cursor
-- 
2.22.0

